src/state/markets/fifo.rs

```diff
@@ -726,7 +777,14 @@ impl<
             }
 
             (
-                FIFORestingOrder::new(trader_index as u64, order_packet.num_base_lots()),
+                FIFORestingOrder::new(
+                    trader_index as u64,
+                    order_packet.num_base_lots(),
+                    order_packet.get_last_valid_slot().unwrap_or(0),
```

Comment:
weak opinion: feel like the constructor should just take in `Option` and handle this